src/python/pants/backend/python/subsystems/pytest.py

```diff
@@ -164,6 +166,16 @@ class PyTest(PythonToolBase):
             """
         ),
     )
+    xdist_enabled = BoolOption(
+        default=False,
```

Comment:
If we temporarily don't consider deprecation policy: thoughts on what default would make sense?

Reply:
I think `false` is a safer default (for example, Django's test setup will step all over itself during DB access if run via `xdist` without special configuration)

Reply:
Also if I try to dogfood this and run in the `pants` repo itself, it fails. Not sure if that means something is broken with the impl or `pants` is missing some config for `xdist`-friendliness 🤔 

Running this:
```
./pants --pytest-xdist-enabled=True test src/python/pants/backend/python/goals/pytest_runner_integration_test.py
```

Produces:
```
============================= test session starts ==============================
gw0 I / gw1 I / gw2 I / gw3 I / gw4 I / gw5 I / gw6 I / gw7 I / gw8 I / gw9 I / gw10 I / gw11 I / gw12 I / gw13 I / gw14 I / gw15 I
INTERNALERROR> def worker_internal_error(self, node, formatted_error):
INTERNALERROR>         """
INTERNALERROR>         pytest_internalerror() was called on the worker.
INTERNALERROR>
INTERNALERROR>         pytest_internalerror() arguments are an excinfo and an excrepr, which can't
INTERNALERROR>         be serialized, so we go with a poor man's solution of raising an exception
INTERNALERROR>         here ourselves using the formatted message.
INTERNALERROR>         """
INTERNALERROR>         self._active_nodes.remove(node)
INTERNALERROR>         try:
INTERNALERROR> >           assert False, formatted_error
INTERNALERROR> E           AssertionError: Traceback (most recent call last):
INTERNALERROR> E             File "/Users/dan.moran/.cache/pants/named_caches/pex_root/venvs/s/b039cb65/venv/lib/python3.7/site-packages/_pytest/main.py", line 266, in wrap_session
INTERNALERROR> E               config.hook.pytest_sessionstart(session=session)
INTERNALERROR> E             File "/Users/dan.moran/.cache/pants/named_caches/pex_root/venvs/s/b039cb65/venv/lib/python3.7/site-packages/pluggy/_hooks.py", line 265, in __call__
INTERNALERROR> E               return self._hookexec(self.name, self.get_hookimpls(), kwargs, firstresult)
INTERNALERROR> E             File "/Users/dan.moran/.cache/pants/named_caches/pex_root/venvs/s/b039cb65/venv/lib/python3.7/site-packages/pluggy/_manager.py", line 80, in _hookexec
INTERNALERROR> E               return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
INTERNALERROR> E             File "/Users/dan.moran/.cache/pants/named_caches/pex_root/venvs/s/b039cb65/venv/lib/python3.7/site-packages/pluggy/_callers.py", line 60, in _multicall
INTERNALERROR> E               return outcome.get_result()
INTERNALERROR> E             File "/Users/dan.moran/.cache/pants/named_caches/pex_root/venvs/s/b039cb65/venv/lib/python3.7/site-packages/pluggy/_result.py", line 60, in get_result
INTERNALERROR> E               raise ex[1].with_traceback(ex[2])
INTERNALERROR> E             File "/Users/dan.moran/.cache/pants/named_caches/pex_root/venvs/s/b039cb65/venv/lib/python3.7/site-packages/pluggy/_callers.py", line 39, in _multicall
INTERNALERROR> E               res = hook_impl.function(*args)
INTERNALERROR> E             File "/private/var/folders/4r/y3nczt9d6nbg25wvwxbqg3t00000gp/T/pants-sandbox-TGT0J8/src/python/pants/conftest.py", line 37, in pytest_sessionstart
INTERNALERROR> E               f"In order for `pants` to be a namespace package, {namespace_init_path} must not "
INTERNALERROR> E           Exception: In order for `pants` to be a namespace package, src/python/pants/__init__.py must not exist on disk. See the explanation in /private/var/folders/4r/y3nczt9d6nbg25wvwxbqg3t00000gp/T/pants-sandbox-TGT0J8/src/python/pants/conftest.py.
INTERNALERROR>
INTERNALERROR> /Users/dan.moran/.cache/pants/named_caches/pex_root/venvs/s/b039cb65/venv/lib/python3.7/site-packages/xdist/dsession.py:192: AssertionError
[gw0] node down: Not properly terminated

replacing crashed worker gw0
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/Users/dan.moran/.cache/pants/named_caches/pex_root/venvs/s/b039cb65/venv/lib/python3.7/site-packages/_pytest/main.py", line 268, in wrap_session
INTERNALERROR>     session.exitstatus = doit(config, session) or 0
INTERNALERROR>   File "/Users/dan.moran/.cache/pants/named_caches/pex_root/venvs/s/b039cb65/venv/lib/python3.7/site-packages/_pytest/main.py", line 322, in _main
INTERNALERROR>     config.hook.pytest_runtestloop(session=session)
INTERNALERROR>   File "/Users/dan.moran/.cache/pants/named_caches/pex_root/venvs/s/b039cb65/venv/lib/python3.7/site-packages/pluggy/_hooks.py", line 265, in __call__
INTERNALERROR>     return self._hookexec(self.name, self.get_hookimpls(), kwargs, firstresult)
INTERNALERROR>   File "/Users/dan.moran/.cache/pants/named_caches/pex_root/venvs/s/b039cb65/venv/lib/python3.7/site-packages/pluggy/_manager.py", line 80, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
INTERNALERROR>   File "/Users/dan.moran/.cache/pants/named_caches/pex_root/venvs/s/b039cb65/venv/lib/python3.7/site-packages/pluggy/_callers.py", line 60, in _multicall
INTERNALERROR>     return outcome.get_result()
INTERNALERROR>   File "/Users/dan.moran/.cache/pants/named_caches/pex_root/venvs/s/b039cb65/venv/lib/python3.7/site-packages/pluggy/_result.py", line 60, in get_result
INTERNALERROR>     raise ex[1].with_traceback(ex[2])
INTERNALERROR>   File "/Users/dan.moran/.cache/pants/named_caches/pex_root/venvs/s/b039cb65/venv/lib/python3.7/site-packages/pluggy/_callers.py", line 39, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/Users/dan.moran/.cache/pants/named_caches/pex_root/venvs/s/b039cb65/venv/lib/python3.7/site-packages/xdist/dsession.py", line 117, in pytest_runtestloop
INTERNALERROR>     self.loop_once()
INTERNALERROR>   File "/Users/dan.moran/.cache/pants/named_caches/pex_root/venvs/s/b039cb65/venv/lib/python3.7/site-packages/xdist/dsession.py", line 140, in loop_once
INTERNALERROR>     call(**kwargs)
INTERNALERROR>   File "/Users/dan.moran/.cache/pants/named_caches/pex_root/venvs/s/b039cb65/venv/lib/python3.7/site-packages/xdist/dsession.py", line 229, in worker_errordown
INTERNALERROR>     self._active_nodes.remove(node)
INTERNALERROR> KeyError: <WorkerController gw0>

```

Reply:
I think our conftest.py has a problem.

But yeah, sounds good on False being the default! I agree with you.